libnavigation-ui/src/main/java/com/mapbox/navigation/ui/VoiceInstructionCache.java

```diff
@@ -28,32 +28,37 @@
     this.connectivityStatus = connectivityStatus;
   }
 
-  void preCache(DirectionsRoute route) {
-    if (!connectivityStatus.isConnected()) {
+  void initCache(DirectionsRoute route) {
+    totalVoiceInstructions = 0;
+
+    List<RouteLeg> routeLegs = route.legs();
+    if (routeLegs == null) {
       return;
     }
 
-    totalVoiceInstructions = 0;
-    currentVoiceInstructionsCachedIndex = 0;
-    isVoiceInstructionsToCacheThresholdReached = false;
-    for (int i = 0; i < route.legs().size(); i++) {
-      RouteLeg leg = route.legs().get(i);
-      for (int j = 0; j < leg.steps().size(); j++) {
-        LegStep step = leg.steps().get(j);
-        for (VoiceInstructions ignored : step.voiceInstructions()) {
-          totalVoiceInstructions++;
-        }
+    for (int i = 0; i < routeLegs.size(); i++) {
+      RouteLeg leg = routeLegs.get(i);
+      if (leg == null) {
```

Comment:
Do we need all these `null` checks? We should 👀 if they're nullable because in previous implementation we were not checking 🤔

Reply:
yes, technically they are nullable. and we do have compile warnings about the NPE possibility.
In reality, I don't know if such route that has those null attributes is possible or not. Unless the API side can make sure they always return nonnull attributes, but in that case, they should mart it in their interface or entity class declaration.

Reply:
There's this ticket in `mapbox-java` https://github.com/mapbox/mapbox-java/issues/1087 trying to do that exercise. Could you check with @mapbox/navigation-api?

Reply:
cool, I can open a ticket to check with them.

do you think we don't need those null check in this PR before they get back to us? or we could merge this first, then update it on basis of their response? I am ok with either option.

Reply:
Is this because of the fact that in the core SDK, `RouteProgress` and every object inside it is `@Nullable` now? If so, we need to fix `RouteProgress` upstream and mark all of these fields accordingly based on if they're required or not. We can take a look at this in a separate PR. Could you cut a follow up ticket so we don't forget 🙏?